vcf.c

```diff
@@ -2362,7 +2396,23 @@ static int vcf_parse_format(kstring_t *s, const bcf_hdr_t *h, bcf1_t *v, char *p
                 int32_t *x = (int32_t*)(z->buf + z->size * m);
                 for (l = 0;; ++t) {
                     if (*t == '.') x[l++] = bcf_int32_missing, ++t; // ++t to skip "."
-                    else x[l++] = strtol(t, &t, 10);
+                    else    // x[l++] = strtol(t, &t, 10);
+                    {
+                        errno = 0;
+                        char *te;
+                        long int tmp_val = strtol(t, &te, 10);
```

Comment:
Should be

```
int64_t tmp_val = strtoll(t &te, 10);
```

so 32-bit systems are still doing the check with64-bit quantities.  This is how code elsewhere works.  (Eg see the v->pos assignment).

(I can amend this myself though.)

Reply:
For FORMAT fields I kept and agree with the original @daviesrob's approach of not supporting 64-bit values. We try to express FORMAT fields using int8_t whenever possible because this is the size bottleneck for VCF, would not like to encourage 64-bit values there at all.

(Having said that, of course, there is the PS tag which can be any number but usually would refer to a genomic coordinate and therefore could be 64-bit in principle. However, this can wait as it is not in widespread use for human data, not speaking about the hypothetical large genomes.)

Reply:
Ok fair enough.  I don't know what's supported and what isn't, but if it's just permitted as 32-bit only then it's fine.  Thanks.